docs/mergeop.md

```diff
@@ -0,0 +1,439 @@
+# MergeOp Requirements
```

Comment:
It shouldn't be implementation notes(these can live in PRs) but structured description of
- what it is
- how to use it
- how it is implemented and what code locations are important

Eg. starting from the top "MergeOp Requirements" is not something a new reader would understand. What requirements?

It should describe the current state firmly, not "how things should work". It's documentation for the existing thing, not a proposal or brainstorm on the topic. Another way to look at it would be if this was a blogpost "Understanding MergeOp internals" - you probably wouldn't start it like this then.

Reply:
Cool, I'll work on updating it to fit your description better and send out a separate PR.